src/System.Runtime.Analyzers/Core/SpecifyStringComparison.cs

```diff
@@ -24,15 +29,117 @@ public abstract class SpecifyStringComparisonAnalyzer : DiagnosticAnalyzer
+                var currentCulture = stringComparisonType.GetMembers("CurrentCulture").SingleOrDefault();
+                var currentCultureIgnoreCase = stringComparisonType.GetMembers("CurrentCultureIgnoreCase").SingleOrDefault();
+                var ordinalIgnoreCase = stringComparisonType.GetMembers("OrdinalIgnoreCase").SingleOrDefault();
+                var ordinal = stringComparisonType.GetMembers("Ordinal").SingleOrDefault();
+
+                // Without these symbols the rule cannot run
+                if (stringComparisonType == null || stringType == null || currentCulture == null ||
+                    currentCultureIgnoreCase == null || ordinalIgnoreCase == null || ordinal == null)
+                {
+                    return;
+                }
+
+                var stringCompareToNamedMethods = stringType.GetMembers("CompareTo").OfType<IMethodSymbol>();
+                var stringCompareToParameterString = stringCompareToNamedMethods.GetSingleOrDefaultMemberWithName("string.CompareTo(string)");
+                var stringCompareToParameterObject = stringCompareToNamedMethods.GetSingleOrDefaultMemberWithName("string.CompareTo(object)");
+
+                var boolString = csaContext.Compilation.Language == LanguageNames.CSharp ? "bool" : "Boolean";
+                var intString = csaContext.Compilation.Language == LanguageNames.CSharp ? "int" : "Integer";
+                var stringCompareNamedMethods = stringType.GetMembers("Compare").OfType<IMethodSymbol>();
+                var stringCompareParameterStringStringBool = stringCompareNamedMethods.GetSingleOrDefaultMemberWithName($"string.Compare(string, string, {boolString})");
+                var stringCompareParameterStringStringComparison = stringCompareNamedMethods.GetSingleOrDefaultMemberWithName("string.Compare(string, string, System.StringComparison)");
+                var stringCompareParameterStringIntStringIntIntBool = stringCompareNamedMethods.GetSingleOrDefaultMemberWithName($"string.Compare(string, {intString}, string, {intString}, {intString}, {boolString})");
+                var stringCompareParameterStringIntStringIntIntComparison = stringCompareNamedMethods.GetSingleOrDefaultMemberWithName($"string.Compare(string, {intString}, string, {intString}, {intString}, System.StringComparison)");
+
+                IDictionary<IMethodSymbol, IMethodSymbol> overloadMap = new Dictionary<IMethodSymbol, IMethodSymbol>();
+                overloadMap.AddKeyValueIfNotNull(stringCompareToParameterString, stringCompareParameterStringStringComparison);
+                overloadMap.AddKeyValueIfNotNull(stringCompareToParameterObject, stringCompareParameterStringStringComparison);
+                overloadMap.AddKeyValueIfNotNull(stringCompareParameterStringStringBool, stringCompareParameterStringStringComparison);
+                overloadMap.AddKeyValueIfNotNull(stringCompareParameterStringIntStringIntIntBool, stringCompareParameterStringIntStringIntIntComparison);
+
```

Comment:
There are other ways to generate diagnostics, so we should not bail. This `overloadMap` is for some predefined member behavior. We still need to do the check for generic behavior